src/drive/mobile/modules/mediaBackup/duck/bugFix.js

```diff
@@ -0,0 +1,206 @@
+import { getTranslateFunction } from 'drive/mobile/lib/i18n'
+import {
+  getReferencedFolders,
+  REF_PHOTOS,
+  getOrCreateFolderWithReference
+} from 'folder-references'
+import { CozyFile } from 'models'
+
+export const fixMagicFolderName = async (client, savedFromMyDeviceFolder) => {
```

Comment:
This function is very long and hard to follow with many nested if/else. I think we can rewrite it with the following structure

1. Make sure the parent folder is OK. Rename it if necessary, make sure it has the correct reference, and cleaning up old folder names.
2. Rename savedFromMyDeviceFolder, move it to the folder from step 1, and clean up if needed

FWIW here's the file as I hacked along it https://gist.github.com/y-lohse/bf77db3860ba46b2bea1fe83c67a6b16